### PR TITLE
Modified mentee registration validations

### LINKED ISSUE #128: Contact number is not being validated in Mentee Application Form
**Describe the bug**

**To Reproduce**
1. Go to Home Page.
2. Click on 'Find a Mentor'.
3. Click on one of the available mentor profiles.
4. Click on the "Apply" button.
5. On the mentee application page, locate the contact number input field.
6. Enter an invalid contact number (e.g., a number that is too long).
7. Click 'Next.'  

**Expected behavior**

The contact number input field should validate the input using a suitable regex pattern. If an invalid contact number is entered, the system should display a proper error message instructing the mentee to reenter a valid contact number. The mentee should not be able to proceed to the next step until a valid contact number is provided.

**Screenshots**

contact

**Desktop (please complete the following information):**
 - OS: Linux Mint 21.3
 - Browser Firefox
 - Version 128.0

**Additional context**

**Are you working on it?**
No

### PR DESCRIPTION
**Pull request by Disura-Randunu**
## Purpose

The purpose of this PR is to fix #128

## Goals

Improve validations on Mentee registration.

## Approach

Modified Mentee Application Validation Schema to handle possible invalids for several fields (contact no, graduated year, year of study. cv link)

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/schemas.ts [hunk 1]
```diff
@@ -4,7 +4,10 @@ export const MenteeApplicationSchema = z.object({
   firstName: z.string().min(1, { message: 'First name cannot be empty' }),
   lastName: z.string().min(1, { message: 'Last name cannot be empty' }),
   email: z.string().email({ message: 'Invalid email address' }),
-  contactNo: z.string().min(1, { message: 'Contact number cannot be empty' }),
+  contactNo: z
+    .string()
+    .min(1, { message: 'Contact number cannot be empty' })
+    .regex(/^\+[1-9]\d{1,14}$/, 'Invalid contact number'),
```
Comment (anjula-sack) [inline src/schemas.ts]: @Disura-Randunu Did you check this for multiple countries? Each country has different lengths.

Comment (Disura-Randunu) [reply src/schemas.ts]: Thats why a common length was added. Around 14 digits. Should start with + to indicate the country code

Comment (anjula-sack) [reply src/schemas.ts]: We probably need to say that you need to add + then let's remove + validations

Comment (Disura-Randunu) [reply src/schemas.ts]: /^\+[1-9]\d{1,3}\d{6,12}$/
this is a bit more correct than before. can have upto 3 numbers after + which makes the country code. then 6-12 digits which is kind of standard amount of numbers in any country.

Comment (Disura-Randunu) [reply src/schemas.ts]: @anjula-sack updated contactNo validation

**Review by anjula-sack — APPROVED** [review]
Thank you! @Disura-Randunu

**Pull request merged**